# 00 - Data collection

The five files in `data/raw/` are 99acres.com scrapes of Gurgaon-area listings, from CampusX's DSMP real-estate capstone. No scraping code was provided. Nothing is processed here - this notebook just documents the inputs.

In [1]:
from pathlib import Path
import pandas as pd
from IPython.display import display

REPO_ROOT = Path.cwd().parents[0] if Path.cwd().name == "notebooks" else Path.cwd()
RAW = REPO_ROOT / "data" / "raw"

RAW_FILES = [
    "flats.csv",
    "houses.csv",
    "independent-house.csv",
    "residential-land.csv",
    "appartments.csv",
]


## The raw files

| file | what it is |
|---|---|
| `flats.csv` | individual flat/apartment listings |
| `houses.csv` | individual independent-house listings |
| `independent-house.csv` | the **same scrape** as `houses.csv` (shared `property_id`s), with 51 extra raw rows |
| `residential-land.csv` | plot / residential-land listings |
| `appartments.csv` | project-level records (per-BHK price bands, amenities, landmark distances) — one row per development, not per listing |

In [2]:
summary = []
for name in RAW_FILES:
    df = pd.read_csv(RAW / name)
    summary.append({"file": name, "rows": df.shape[0], "cols": df.shape[1]})
pd.DataFrame(summary).set_index("file")


,rows,cols
file,,
flats.csv,3017,20
houses.csv,1044,21
independent-house.csv,1095,21
residential-land.csv,2422,14
appartments.csv,247,7


In [3]:
for name in RAW_FILES:
    df = pd.read_csv(RAW / name)
    print(f"\n=== {name}  {df.shape} ===")
    print("columns:", list(df.columns))
    display(df.head(2))



=== flats.csv  (3017, 20) ===
columns: ['property_name', 'link', 'society', 'price', 'area', 'areaWithType', 'bedRoom', 'bathroom', 'balcony', 'additionalRoom', 'address', 'floorNum', 'facing', 'agePossession', 'nearbyLocations', 'description', 'furnishDetails', 'features', 'rating', 'property_id']


,property_name,link,society,price,area,areaWithType,bedRoom,bathroom,balcony,additionalRoom,address,floorNum,facing,agePossession,nearbyLocations,description,furnishDetails,features,rating,property_id
0,2 BHK Flat in Krishna Colony,https://www.99acres.com/2-bhk-bedroom-apartmen...,maa bhagwati residency,45 Lac,"₹ 5,000/sq.ft.",Carpet area: 900 (83.61 sq.m.),2 Bedrooms,2 Bathrooms,1 Balcony,NaN,"Krishna Colony, Gurgaon, Haryana",4th of 4 Floors,West,1 to 5 Year Old,"['Chintapurni Mandir', 'State bank ATM', 'Pear...",So with lift.Maa bhagwati residency is one of ...,"['3 Fan', '4 Light', '1 Wardrobe', 'No AC', 'N...","['Feng Shui / Vaastu Compliant', 'Security / F...","['Environment4 out of 5', 'Safety4 out of 5', ...",C68850746
1,2 BHK Flat in Ashok Vihar,https://www.99acres.com/2-bhk-bedroom-apartmen...,Apna Enclave,50 Lac,"₹ 7,692/sq.ft.",Carpet area: 650 (60.39 sq.m.),2 Bedrooms,2 Bathrooms,1 Balcony,NaN,"46b, Ashok Vihar, Gurgaon, Haryana",1st of 3 Floors,West,10+ Year Old,"['Chintapurni Mandir', 'Sheetla Mata Mandir', ...","Property situated on main road, railway statio...","['3 Wardrobe', '4 Fan', '1 Exhaust Fan', '1 Ge...","['Security / Fire Alarm', 'Maintenance Staff',...","['Environment4 out of 5', 'Safety4 out of 5', ...",H68850564



=== houses.csv  (1044, 21) ===
columns: ['property_name', 'link', 'society', 'price', 'rate', 'area', 'areaWithType', 'bedRoom', 'bathroom', 'balcony', 'additionalRoom', 'address', 'noOfFloor', 'facing', 'agePossession', 'nearbyLocations', 'description', 'furnishDetails', 'features', 'rating', 'property_id']


,property_name,link,society,price,rate,area,areaWithType,bedRoom,bathroom,balcony,...,address,noOfFloor,facing,agePossession,nearbyLocations,description,furnishDetails,features,rating,property_id
0,5 Bedroom House for sale in Sector 70A Gurgaon,https://www.99acres.com/5-bhk-bedroom-independ...,Bptp Visionnaire,5.25 Crore,"₹ 20,115/sq.ft.",(242 sq.m.) Plot Area,Plot area 290(242.48 sq.m.),5 Bedrooms,4 Bathrooms,3+ Balconies,...,"29b, Sector 70A Gurgaon, Gurgaon, Haryana",3 Floors,North-East,0 to 1 Year Old,"['Good Earth City Center 2', 'Kunskapsskolan I...",Do you wish to buy an independent house in sec...,"['1 Wardrobe', '1 Fan', '1 Exhaust Fan', '1 Ge...","['Feng Shui / Vaastu Compliant', 'Private Gard...","['Environment5 out of 5', 'Lifestyle4 out of 5...",K70037724
1,5 Bedroom House for sale in Sector 21A Faridabad,https://www.99acres.com/5-bhk-bedroom-independ...,NaN,5.7 Crore,"₹ 105,751/sq.ft.",(50 sq.m.) Plot Area,Plot area 539(50.07 sq.m.),5 Bedrooms,4 Bathrooms,2 Balconies,...,"Sector 21A Faridabad, Gurgaon, Haryana",2 Floors,NaN,5 to 10 Year Old,NaN,"Hi, we have an independent house/villa availab...","['1 Water Purifier', '5 Fan', '1 Exhaust Fan',...","['Private Garden / Terrace', 'Park', 'Visitor ...",NaN,E69288322



=== independent-house.csv  (1095, 21) ===
columns: ['property_name', 'link', 'society', 'price', 'rate', 'area', 'areaWithType', 'bedRoom', 'bathroom', 'balcony', 'additionalRoom', 'address', 'noOfFloor', 'facing', 'agePossession', 'nearbyLocations', 'description', 'furnishDetails', 'features', 'rating', 'property_id']


,property_name,link,society,price,rate,area,areaWithType,bedRoom,bathroom,balcony,...,address,noOfFloor,facing,agePossession,nearbyLocations,description,furnishDetails,features,rating,property_id
0,5 Bedroom House for sale in Sector 70A Gurgaon,https://www.99acres.com/5-bhk-bedroom-independ...,Bptp Visionnaire,5.25 Crore,"₹ 20,115/sq.ft.",(242 sq.m.) Plot Area,Plot area 290(242.48 sq.m.),5 Bedrooms,4 Bathrooms,3+ Balconies,...,"29b, Sector 70A Gurgaon, Gurgaon, Haryana",3 Floors,North-East,0 to 1 Year Old,"['Good Earth City Center 2', 'Kunskapsskolan I...",Do you wish to buy an independent house in sec...,"['1 Wardrobe', '1 Fan', '1 Exhaust Fan', '1 Ge...","['Feng Shui / Vaastu Compliant', 'Private Gard...","['Environment5 out of 5', 'Lifestyle4 out of 5...",K70037724
1,5 Bedroom House for sale in Sector 21A Faridabad,https://www.99acres.com/5-bhk-bedroom-independ...,NaN,5.7 Crore,"₹ 105,751/sq.ft.",(50 sq.m.) Plot Area,Plot area 539(50.07 sq.m.),5 Bedrooms,4 Bathrooms,2 Balconies,...,"Sector 21A Faridabad, Gurgaon, Haryana",2 Floors,NaN,5 to 10 Year Old,NaN,"Hi, we have an independent house/villa availab...","['1 Water Purifier', '5 Fan', '1 Exhaust Fan',...","['Private Garden / Terrace', 'Park', 'Visitor ...",NaN,E69288322



=== residential-land.csv  (2422, 14) ===
columns: ['property_name', 'link', 'society', 'price', 'areaWithType', 'address', 'floorNumAllowed', 'noOfOpenSides', 'possession', 'nearbyLocations', 'description', 'features', 'rating', 'property_id']


,property_name,link,society,price,areaWithType,address,floorNumAllowed,noOfOpenSides,possession,nearbyLocations,description,features,rating,property_id
0,Residential land / Plot in Sector-33 Sohna,https://www.99acres.com/residential-land-plot-...,Central Park Flower Valley Mikasa Plots,3.75 Crore,(209 sq.m.) Plot Area,"Sector-33 Sohna, Gurgaon, Haryana",NaN,1.0,NaN,"['Rapid Metro Sector 55-56', 'CBR Cricket Grou...",East facing 24 meter road plot. Very good loca...,"['Feng Shui / Vaastu Compliant', 'Maintenance ...",NaN,U69990808
1,Residential land / Plot in Sector 93 Gurgaon,https://www.99acres.com/residential-land-plot-...,ROF Insignia Park,2.23 Crore,(17 sq.m.) Plot Area,"Sector 93, Sector 93 Gurgaon, Gurgaon, Haryana",NaN,1.0,undefined,"['Newtown Square Mall', 'Dwarka Expy', 'Yaduva...",At the maine enterance rof insignia park secto...,NaN,NaN,A69982942



=== appartments.csv  (247, 7) ===
columns: ['PropertyName', 'PropertySubName', 'NearbyLocations', 'LocationAdvantages', 'Link', 'PriceDetails', 'TopFacilities']


,PropertyName,PropertySubName,NearbyLocations,LocationAdvantages,Link,PriceDetails,TopFacilities
0,Smartworld One DXP,"2, 3, 4 BHK Apartment in Sector 113, Gurgaon","['Bajghera Road', 'Palam Vihar Halt', 'DPSG Pa...","{'Bajghera Road': '800 Meter', 'Palam Vihar Ha...",https://www.99acres.com/smartworld-one-dxp-sec...,"{'2 BHK': {'building_type': 'Apartment', 'area...","['Swimming Pool', 'Salon', 'Restaurant', 'Spa'..."
1,M3M Crown,"3, 4 BHK Apartment in Sector 111, Gurgaon","['DPSG Palam Vihar Gurugram', 'The NorthCap Un...","{'DPSG Palam Vihar Gurugram': '1.4 Km', 'The N...",https://www.99acres.com/m3m-crown-sector-111-g...,"{'3 BHK': {'building_type': 'Apartment', 'area...","['Bowling Alley', 'Mini Theatre', 'Manicured G..."


### `houses.csv` and `independent-house.csv` are one scrape

**Finding:** every real
`property_id` in `houses.csv` also appears in `independent-house.csv` and vice
versa — `independent-house.csv` just has more raw rows. Merging them as if
additive would double-count ~1,000 properties. The pipeline uses `houses.csv`.

Counting **non-null** ids (both files have some blank `property_id` rows —
8 and 59 respectively):

In [4]:
h_ids = pd.read_csv(RAW / "houses.csv")["property_id"].dropna()
ih_ids = pd.read_csv(RAW / "independent-house.csv")["property_id"].dropna()
h, ih = set(h_ids), set(ih_ids)
print(f"houses.csv           : {len(h_ids)} non-null property_id rows, {len(h)} distinct")
print(f"independent-house.csv : {len(ih_ids)} non-null property_id rows, {len(ih)} distinct")
print(f"distinct ids in common          : {len(h & ih)}")
print(f"distinct ids only in houses.csv : {len(h - ih)}")
print(f"distinct ids only in indep-house: {len(ih - h)}")


houses.csv           : 1036 non-null property_id rows, 1022 distinct
independent-house.csv : 1036 non-null property_id rows, 1022 distinct
distinct ids in common          : 1022
distinct ids only in houses.csv : 0
distinct ids only in indep-house: 0


## Compliance note

> Scraping 99acres.com sits in a legal gray area. Public derivatives of this
> project typically carry an **"educational purposes only"** disclaimer for
> exactly that reason — worth adding the same to the eventual README, and worth
> a second look before committing the raw scraped CSVs to a public repo.

This project treats the data as **educational / non-commercial only**.

## Available but not yet integrated

`02`–`07` run on `flats.csv` + `houses.csv` only.

- **`residential-land.csv` (2,422 rows) — genuinely unused.** Different schema
  (`areaWithType`, `floorNumAllowed`, `noOfOpenSides`, `possession`; no
  `bedRoom` / `bathroom` / `furnishDetails`), so integrating it means a
  parallel cleaning + modelling track, not a column add. The main
  differentiation opportunity left open is **price prediction on land / plot
  data**, which none of the 16 original notebooks touch. Left for later; noted here so the raw file isn't forgotten.
- **`independent-house.csv` — effectively already integrated.** Same 21-column
  schema and (per the check above) the same 1,022 properties as `houses.csv`;
  its 51 extra raw rows are mostly the blank-`property_id` kind. Not a separate
  data source to bring in.